spring-cloud-stream-binder-jms-common/src/main/java/org/springframework/cloud/stream/binder/jms/utils/SpecCompliantJmsHeaderMapper.java

```diff
@@ -0,0 +1,42 @@
+package org.springframework.cloud.stream.binder.jms.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.jms.Message;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.integration.jms.DefaultJmsHeaderMapper;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * Replace all header names that contain '-' with '_' due to IBM MQ header name
+ * constraints.
+ *
+ * See http://stackoverflow.com/a/30024766/2408961 for context.
+ *
+ * @author Donovan Muller
+ */
+public class SpecCompliantJmsHeaderMapper extends DefaultJmsHeaderMapper {
```

Comment:
I don't entirely disagree, but it would be convenient for the framework to provide it, at least as an option - consider the use case when bridging, say, rabbitmq to jms.